errata-email-notifications.py

```diff
@@ -0,0 +1,247 @@
+                logging.warning('Skipping notifications for %s as it looks like ' \
+                                'is the first time this script is running', dist)
+                continue
+
+            with open(os.path.join(BASEPATH, last_processed_ts_file), 'r') as f:
+                last_processed_ts = f.readline()
+            # Only take those whose ts is bigger than last_processed_ts
+            new_erratas = [
+                x for x in errata_data
+                if x['updated_date']['$date'] > int(last_processed_ts)
+            ]
+            logging.debug('Found %d new erratas for %s distribution',
+                len(new_erratas), dist
+            )
+
+            # We want to start sending notifications from the oldest one
+            for errata in reversed(new_erratas):
+                # TODO: Make an errata class so retrieving info in the way
+                # we need is more comfortable for our purposes
+                subject = SUBJECT_TEMPLATE.substitute(
+                    errata_type=errata['type'].capitalize(),
+                    # TODO: Add some checks to ensure that we include
+                    # the errata_id at the beginning of the title.
+                    # Apparently, there's no strict format, we just copy
+                    # the errata info from RHEL and sometimes it differs
+                    errata_title=errata['title']
+                )
+                content = CONTENT_TEMPLATE.substitute(
+                    errata_type=errata['type'].capitalize(),
+                    errata_severity=errata['severity'].capitalize(),
+                    errata_date=self.ts_to_date(errata['updated_date']['$date']),
+                    errata_description=errata['description'],
+                    errata_link=self.get_almalinux_errata_href(errata)
+                )
+
+                msg = EmailMessage()
+                msg['From'] = f'{EMAIL_FROM_NAME} <{self.sender}>'
+                msg['To'] = self.recipient
+                msg['Subject'] = subject
+                msg.set_content(content)
+
+                try:
+                    self.smtp_session.send(msg)
+                    self.save_last_processed_ts(last_processed_ts_file,
+                                                errata['updated_date']['$date'])
+                except:
+                    logging.error('Could not send errata notification email')
+
+        self.smtp_session.close()
+
+
+    def fetch_errata_data(self, url):
+        logging.debug('Fetching errata data from %s', url)
+        # TODO: If this script requires a considerable allocation of memory, we can
+        # figure out if we can publish the errata json files sorted by date.
+        # This way we can think about reading the response in chunks and stop reading
+        # when we reach the last_processed_ts. This could save some memory allocation
+        # during the execution of the script.
+        try:
+            response = request.urlopen(url)
```

Comment:
Please, use library `requests`. It better handles http requests and can process json (and other) response.

Reply:
Done